include/aws/common/byte_order.inl

```diff
@@ -11,6 +11,7 @@
 
 #ifdef _WIN32
 #    include <stdlib.h>
+#    include <winsock.h>
```

Comment:
We'll need to take a different approach.

This `.inl` file is included from our public `.h` files, and it's basically a computer crime to include a Windows header from a public header. Due to the sloppy naming infecting all code that includes them.

We'll need to move this function implemention to a normal .c file and do the include from there, at least for windows